src/lightning/pytorch/loops/fit_loop.py

```diff
@@ -302,14 +314,92 @@ def setup_data(self) -> None:
                 category=PossibleUserWarning,
             )
 
+    @property
+    def restarted_on_epoch_start(self) -> bool:
```

Comment:
```suggestion
    def is_restarted_on_epoch_start(self) -> bool:
```
how about making the name easier to understand it is bool

Reply:
I took it from `self.restarting`, if we feel strongly then I can change it in a follow up